packages/fuels/tests/bindings/sharing_types/contract_b/src/main.sw

```diff
@@ -0,0 +1,40 @@
+contract;
+use shared_lib::*;
+
+struct UniqueStructToContractB<T> {
+    a: T,
+}
+
+struct StructSameNameButDifferentInternals {
+    a: u64,
+}
+
+enum EnumSameNameButDifferentInternals {
+    a: u64,
+}
```

Comment:
So I know they are technically different internals, but I think the test would be better if the type were for instance a `&str`. Otherwise in the test above we aren't really convinced that the internals are actually different.

Reply:
Hmm. Something that isn't easily convertible from one type to the other. Good point. I'll change it now.

Reply:
But wait. Rust doesn't accept implicit u32 conversions to u64 and vice-versa. The same-named types have `u32` and `u64` for their `a` field/variant.

But still, even though the literals are given explicitly as `u32` and `u64` literals, somebody might still edit the test without being explicit about the literals and eventually, should abigen fail to generate the types properly, might cause the test to produce a false negative.

I'll change it to be an array or something. Strings are a hassle.

Reply:
Done.